lib/Conversion/ExportVerilog/ExportVerilogInternals.h

```diff
@@ -238,6 +238,16 @@ struct SharedEmitterState {
   // Emitter options extracted from the top-level module.
   const LoweringOptions &options;
 
+  /// This keeps track of prefixes assigned to enum types. The key may be any
+  /// type - this implies that enum types which alias in their fields, but have
+  /// been typedecl'd to different names may be assigned different prefixes.
+  DenseMap<Type, StringAttr> enumPrefixes;
+
+  /// Returns the field name for an enum field of a given enum field attr. In
+  /// case a prefix exists for the provided enum type, the prefix will be
+  /// applied. If not, the raw field name is returned.
+  std::string getEnumFieldName(hw::EnumFieldAttr attr) const;
+
```

Comment:
Can we move these members to `FieldNameResolver` and perform the logic of `gatherEnumPrefixes` in the GlobalnameResolver constructor?

Reply:
Thanks for the suggestion - reimplemented based on `FieldNameResolver`/`GlobalNameResolver`. `FieldNameResolver::getRenamedFieldName` should probably be renamed to `getRenamedStructFieldName`.